# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.h
# Commit: 48fc3f91f020
# Full Hash: 48fc3f91f0207ca38b1b6ab88f33c7c7c999d152
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core"
#   
#   This reverts commit 8d55fdef16bcd3e93e2aab7e4aead025c2d07ae4.
# ==============================================================================

diff -r 260574c2a78e -r 48fc3f91f020 dom/base/RangeUtils.h
--- a/dom/base/RangeUtils.h	Thu May 08 22:53:06 2025 +0300
+++ b/dom/base/RangeUtils.h	Thu May 08 22:53:07 2025 +0300
@@ -11,7 +11,6 @@
 #include "mozilla/RangeBoundary.h"
 #include "nsIContent.h"
 #include "nsINode.h"
-#include "nsContentUtils.h"
 
 namespace mozilla {
 
@@ -56,6 +55,8 @@
   using AbstractRange = dom::AbstractRange;
 
  public:
+  static nsINode* GetParentNodeInSameSelection(const nsINode* aNode);
+
   /**
    * GetRawRangeBoundaryBefore() and GetRawRangeBoundaryAfter() retrieve
    * RawRangeBoundary which points before or after aNode.
@@ -136,9 +137,6 @@
   /**
    * The caller needs to ensure aNode is in the same doc like aAbstractRange.
    */
-  template <TreeKind aKind = TreeKind::ShadowIncludingDOM,
-            typename = std::enable_if_t<aKind == TreeKind::ShadowIncludingDOM ||
-                                        aKind == TreeKind::Flat>>
   static Maybe<bool> IsNodeContainedInRange(nsINode& aNode,
                                             AbstractRange* aAbstractRange);
 
@@ -147,18 +145,12 @@
    * ends after a range.  If neither it is contained inside the range.
    * Note that callers responsibility to ensure node in same doc as range.
    */
-  template <TreeKind aKind = TreeKind::ShadowIncludingDOM,
-            typename = std::enable_if_t<aKind == TreeKind::ShadowIncludingDOM ||
-                                        aKind == TreeKind::Flat>>
   static nsresult CompareNodeToRange(nsINode* aNode,
                                      AbstractRange* aAbstractRange,
                                      bool* aNodeIsBeforeRange,
                                      bool* aNodeIsAfterRange);
 
-  template <TreeKind aKind, typename SPT, typename SRT, typename EPT,
-            typename ERT,
-            typename = std::enable_if_t<aKind == TreeKind::ShadowIncludingDOM ||
-                                        aKind == TreeKind::Flat>>
+  template <typename SPT, typename SRT, typename EPT, typename ERT>
   static nsresult CompareNodeToRangeBoundaries(
       nsINode* aNode, const RangeBoundaryBase<SPT, SRT>& aStartBoundary,
       const RangeBoundaryBase<EPT, ERT>& aEndBoundary, bool* aNodeIsBeforeRange,